File: Laba4/Java/Table.java
import java.util.concurrent.Semaphore;

public class Table {
    private final Semaphore[] forks = new Semaphore[Main.PHILOSOPHERS_COUNT];

    public Table() {
        for (int i = 0; i < forks.length; i++) {
            forks[i] = new Semaphore(1, true); 
        }
    }

    public void getFork(int id) {
        try {
            forks[id].acquire();
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }

    public void putFork(int id) {
        forks[id].release();
    }
}

File: Laba4/Sharp/Table.cs
namespace Laba4
{
    public class Table
    {
        private readonly Semaphore[] forks = new Semaphore[Program.PhilosophersCount];

        public Table()
        {
            for (int i = 0; i < forks.Length; i++)
            {
                forks[i] = new Semaphore(1, 1);
            }
        }

        public void GetFork(int id)
        {
            forks[id].WaitOne();
        }

        public void PutFork(int id)
        {
            forks[id].Release();
        }
    }
}